ophyd/device.py

```diff
@@ -454,7 +454,11 @@ def _read_attr_list(self, attr_list):
         '''Get a 'read' dictionary containing attributes in attr_list'''
         values = OrderedDict()
         for name in attr_list:
-            signal = getattr(self, name)
+            try:
+                signal = getattr(self, name)
+            except AttributeError:
+                # This field may be added by a mixin class.
```

Comment:
I removed the offending commit. So, for now, users must omit `['image']` from the `read_attrs`. It will simply always be read. We can postpone deliberation on how best to address this.